learning/katas/go/Core Transforms/Branching/Branching/task.md

```diff
@@ -0,0 +1,31 @@
+<!--
+  ~ Licensed to the Apache Software Foundation (ASF) under one
+  ~ or more contributor license agreements.  See the NOTICE file
+  ~ distributed with this work for additional information
+  ~ regarding copyright ownership.  The ASF licenses this file
+  ~ to you under the Apache License, Version 2.0 (the
+  ~ "License"); you may not use this file except in compliance
+  ~ with the License.  You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+# Branching
+
+You can use the same PCollection as input for multiple transforms without consuming the input or 
+altering it.
```

Comment:
Not technically true. If the elements are reference values, like maps and pointers and similar, they can be mutated, and the side effects would be seen in other fused DoFns in the bundle.

The recommendation is to never mutate any input to a DoFn, and never mutate/reuse a value after it has been passed to an emitter.